src/sampling.jl

```diff
@@ -101,6 +101,29 @@ sample_ordered!(sampler!, rng::AbstractRNG, a::AbstractArray,
         sampler!(rng, a, wv, x)
     end
 
+"""
+This algorithm generates a sorted sample with replacement by
+adapting the classic result that the cumulative sum of n+1 
+exponentially-distributed random numbers divided by the overall sum 
+(dropping the last) is a sorted sample from a uniform[0,1]
+"""
+function uniform_orderstat_sample!(rng::AbstractRNG, a::AbstractArray, x::AbstractArray)
+    1 == firstindex(a) == firstindex(x) ||
+        throw(ArgumentError("non 1-based arrays are not supported"))
+    Base.mightalias(a, x) &&
+        throw(ArgumentError("output array x must not share memory with input array a"))
+    n = length(a)
+    k = length(x)
+    exp_rands = randexp(rng, k)
+    sorted_rands = accumulate!(+, exp_rands, exp_rands)
```

Comment:
This is not safe. From the docstring of `accumulate!`:
```julia
  │ Warning
  │
  │  Behavior can be unexpected when any mutated argument shares memory with any other argument.
```

Reply:
right, changed in `cumsum`, it is more or less the same performance even with it, I updated the timing in the main comment